### PR TITLE
[Workspace] Disable select data source panel for non dashboard admin in workspace create/edit page

### PR DESCRIPTION
**Pull request by raintygao**
### Description

Hide select data source panel for non dashboard admin or when data source is disabled in workspace create/edit page

## Screenshot

image
image
Non OSD admin

image
image
OSD admin

## Testing the changes

Dashboard admin and data source is enabled
Set username in `opensearchDashboards.dashboardAdmin.users` and `data_source.enabled: true`  in YML, then go to workspace create page or edit page, you will see select data source panel

Non Dashboard admin
Set other username in `opensearchDashboards.dashboardAdmin.users` in YML(If you don't set a value you will still be treated as dashboard admin), then go to workspace create page or edit page, you won't see select data source panel.

Data source disabled
Set `data_source.enabled: false` in YML, then go to workspace create page or edit page, you won't see select data source panel.

## Changelog

- feat: Hide select data source panel for non dashboard admin or when data source is disabled in workspace

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_form/workspace_form.test.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+describe('WorkspaceForm', () => {
+  it('should enable data source panel for dashboard admin', () => {
+    const { getByText } = setup(true);
+
+    expect(getByText('Select Data Sources')).toBeInTheDocument();
+  });
+
+  it('should disable data source panel for non dashboard admin', () => {
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_form/workspace_form.test.tsx:42]:
```suggestion
  it('should not display data source panel for non dashboard admin', () => {
```

File: src/plugins/workspace/public/components/workspace_form/workspace_form.tsx [hunk 2]
```diff
@@ -176,22 +177,27 @@ export const WorkspaceForm = (props: WorkspaceFormProps) => {
...
-        <SelectDataSourcePanel
-          errors={formErrors.selectedDataSources}
-          onChange={setSelectedDataSources}
-          savedObjects={savedObjects}
-          selectedDataSources={formData.selectedDataSources}
-          data-test-subj={`workspaceForm-dataSourcePanel`}
-        />
-      </EuiPanel>
+
+      {/* SelectDataSourcePanel is only enabled for dashboard admin */}
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_form/workspace_form.tsx:181]:
```suggestion
      {/* SelectDataSourcePanel is only visible for dashboard admin */}
```

File: changelogs/fragments/7213.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+fix:
+- Disable select data source panel for non dashboard admin in workspace create/edit page ([#7213](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7213))
```
Comment (ruanyl) [inline changelogs/fragments/7213.yml:2]:
```suggestion
- Hide select data source panel for non dashboard admin in workspace create/edit page ([#7213](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7213))
```

**Review by ruanyl — APPROVED** [review]
Approved with minor comments

**Review by ruanyl — DISMISSED** [review]

File: src/plugins/workspace/public/components/workspace_form/workspace_form.tsx [hunk 4]
```diff
@@ -176,22 +177,27 @@ export const WorkspaceForm = (props: WorkspaceFormProps) => {
...
-          errors={formErrors.selectedDataSources}
-          onChange={setSelectedDataSources}
-          savedObjects={savedObjects}
-          selectedDataSources={formData.selectedDataSources}
-          data-test-subj={`workspaceForm-dataSourcePanel`}
-        />
-      </EuiPanel>
+
+      {/* SelectDataSourcePanel is only visible for dashboard admin */}
+      {isDashboardAdmin && (
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_form/workspace_form.tsx:182]: Do we need to detect if data source is enabled?

Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_form/workspace_form.tsx:182]: When data source is disabled, the list will be empty. Do you think we should hide this panel if data source is disabled?

Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_form/workspace_form.tsx:182]: I think we should, when data source is not enabled, there is not even a data source concept right?

Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_form/workspace_form.tsx]: Updated.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**